taskcluster/fenix_taskgraph/transforms/build.py

```diff
@@ -50,15 +51,16 @@ def add_shippable_secrets(config, tasks):
                 ('mls', '.mls_token'),
             )])
         else:
-            task["run"]["pre-gradlew"] = [[
-                "echo", '"{}"'.format(fake_value), ">", target_file
```

Comment:
Bug: This command didn't work. I saw it when I backported the patch on reference-browser (https://github.com/mozilla-mobile/reference-browser/pull/1034 https://firefox-ci-tc.services.mozilla.com/tasks/YsO0TONlTF27dq5WKo38hw/runs/0/logs/https%3A%2F%2Ffirefox-ci-tc.services.mozilla.com%2Fapi%2Fqueue%2Fv1%2Ftask%2FYsO0TONlTF27dq5WKo38hw%2Fruns%2F0%2Fartifacts%2Fpublic%2Flogs%2Flive.log#L1407)

I looked closely into it and it turns out `shell_quote()`

https://github.com/mozilla-mobile/fenix/blob/f20fb504818726c7391c7b126e89d8a8ffea4381/taskcluster/fenix_taskgraph/job.py#L122

ensures `>` is quoted. This usually avoids security problems. Thus we do want to keep `shell_quote`. That's why I moved the fake token generation into its own script (`write-dummy-secret.py`). It's very similar to `get-secret.py`

Reply:
Thanks for the context!